src/create/createWithOptions.test.ts

```diff
@@ -0,0 +1,247 @@
+	excludeLintJSDoc: false,
+	excludeLintJson: false,
+	excludeLintKnip: false,
+	excludeLintMd: false,
+	excludeLintPackageJson: false,
+	excludeLintPackages: false,
+	excludeLintPerfectionist: false,
+	excludeLintRegex: false,
+	excludeLintSpelling: false,
+	excludeLintStrict: false,
+	excludeLintStylistic: false,
+	excludeLintYml: false,
+	excludeReleases: false,
+	excludeRenovate: false,
+	excludeTests: false,
+	funding: "Test Funding",
+	keywords: ["test", "keywords"],
+	logo: { alt: "Test Alt", src: "test.png" },
+	mode: "create",
+	offline: false,
+	owner: "Test Owner",
+	preserveGeneratedFrom: false,
+	repository: "test-repo",
+	skipAllContributorsApi: false,
+	skipGitHubApi: false,
+	skipInstall: false,
+	skipRemoval: false,
+	skipRestore: false,
+	skipUninstall: false,
+	title: "Test Title",
+};
+
+const mock$ = vi.fn();
+
+vi.mock("execa", () => ({
+	get $() {
+		return mock$;
+	},
+}));
+
+const mockOctokit = new Octokit();
+const github = {
+	auth: "auth-token",
+	octokit: mockOctokit,
+};
+
+vi.mock("../shared/cli/spinners.js", () => ({
+	withSpinner: vi
+		.fn()
+		.mockImplementation(
+			async <Return>(label: string, task: SpinnerTask<Return>) => {
+				return await task();
+			},
+		),
+	withSpinners: vi
+		.fn()
+		.mockImplementation(
+			async (label: string, tasks: [string, SpinnerTask<unknown>][]) => {
+				for (const [, task] of tasks) {
+					await task();
+				}
+			},
+		),
+}));
+
+vi.mock("../steps/writing/writeStructure.js", () => ({
+	writeStructure: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/writeReadme/index.js", () => ({
+	writeReadme: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/finalizeDependencies.js", () => ({
+	finalizeDependencies: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../steps/runCommands.js", () => ({
+	runCommands: vi.fn().mockResolvedValue(null),
+}));
+
+vi.mock("../shared/doesRepositoryExist.js", () => ({
+	doesRepositoryExist: vi.fn().mockResolvedValue(true),
+}));
+
+vi.mock("../steps/initializeGitHubRepository/index.js", () => ({
+	initializeGitHubRepository: vi.fn().mockResolvedValue(true),
+}));
+
+vi.mock("../shared/getGitHubUserAsAllContributor.js", () => ({
+	getGitHubUserAsAllContributor: vi.fn().mockResolvedValue({
+		contributions: ["code", "doc"],
+		name: "Test User",
+	}),
+}));
+
+vi.mock("../steps/addToolAllContributors.js", () => ({
+	addToolAllContributors: vi.fn().mockResolvedValue(null),
+}));
+
+it("calls writeStructure with options", async () => {
+	await writeStructure(options);
+	expect(writeStructure).toHaveBeenCalledWith(options);
+});
+
+it("calls writeReadme with options", async () => {
+	await writeReadme(options);
+	expect(writeReadme).toHaveBeenCalledWith(options);
+});
+
+it("calls writeStructure and writeReadme with options", async () => {
+	await createWithOptions({ github, options });
+
+	expect(writeStructure).toHaveBeenCalledWith(options);
+	expect(writeReadme).toHaveBeenCalledWith(options);
+});
```

Comment:
[Suggestion] I personally generally try not to test cases that are _always_ called. Unless there's a known previous bug or tricky thing about the area of code, I think it's fine to just focus on the logic.

For any test that couldn't reasonably have a _"when"_ in its name, let's delete it.

Reply:
Ok that's fine with me! Does that include the test to make sure the mocked CLI commands give the expected output? I included that because I had seen it elsewhere in another test and wanted to be consistent.

Reply:
Looking back at it I think that you could say that the test looking at those commands is a part of the logic for when sendToGitHub is called, so it makes sense to me know how that fits in.